### PR TITLE
[Storage][DataMovement] Attempt to fix flakey StopOnAllFailures tests (again)

### PR DESCRIPTION
**Pull request by jalauzon-msft**

### CODE REVIEW

File: sdk/storage/Azure.Storage.DataMovement/src/StreamToUriJobPart.cs [hunk 1]
```diff
@@ -238,7 +238,8 @@ internal async Task InitialUploadCall(long blockSize, long expectedlength, bool
             {
                 if (singleCall)
                 {
-                    ReadStreamStorageResourceResult result = await _sourceResource.ReadStreamAsync().ConfigureAwait(false);
+                    ReadStreamStorageResourceResult result = await _sourceResource.ReadStreamAsync(
+                        cancellationToken: _cancellationToken).ConfigureAwait(false);
```
Comment (jalauzon-msft) [inline sdk/storage/Azure.Storage.DataMovement/src/StreamToUriJobPart.cs:242]: This part is actually unrelated but noticed while debugging. Just adding the cancellation token to this call.

File: sdk/storage/Azure.Storage.DataMovement/src/TransferJobInternal.cs [hunk 2]
```diff
@@ -263,14 +263,9 @@ public void DisposeHandlers()
         public async Task TriggerJobCancellationAsync()
         {
             if (!_dataTransfer._state.CancellationTokenSource.IsCancellationRequested)
-            {
-                _dataTransfer._state.TriggerCancellation();
-            }
-
-            // Set CancellationInProgress if we are not already pausing
-            if (StorageTransferStatus.PauseInProgress != _dataTransfer.TransferStatus)
```
Comment (amnguye) [inline sdk/storage/Azure.Storage.DataMovement/src/TransferJobInternal.cs:271]: Did this status message not matter is setting it to paused later on?

Comment (jalauzon-msft) [reply sdk/storage/Azure.Storage.DataMovement/src/TransferJobInternal.cs:271]: This was here to make sure we didn't set it to cancelled if it was Pausing instead. Now that this has been moved into the check to see fi the token has been cancelled, we should not need the check and the pause operation will set PauseInProgress and then cancel the token so this code will not run.

**Review by amnguye — APPROVED** [review]

**Pull request merged**